x-pack/legacy/plugins/maps/public/layers/styles/vector/components/color/color_stops_utils.js

```diff
@@ -14,37 +28,114 @@ export function removeRow(colorStops, index) {
   return [...colorStops.slice(0, index), ...colorStops.slice(index + 1)];
 }
 
-export function addRow(colorStops, index) {
+export function addOrdinalRow(colorStops, index) {
+  return addRow(colorStops, index, COLOR_MAP_TYPE.ORDINAL);
+}
+
+export function addCategoricalRow(colorStops, index) {
+  return addRow(colorStops, index, COLOR_MAP_TYPE.CATEGORICAL);
+}
+
+export function addRow(colorStops, index, colorMapType) {
```

Comment:
How about passing `nextValue` instead of `colorMapType` into `addRow`? That way you get the maximum reuse out of `addRow` without an logical branching.